### ISSUE TITLE
Support /*exported*/ for no-unused-vars

### ISSUE THREAD

**Issue by Skalman** [eslint/eslint#1200]
The `no-unused-vars` option should ignore variables that are explicitly exported with an `/*exported*/` comment. With the option enabled, the following code generates two errors. I expected none (using ESLint 0.7.4).

```js
/*exported myVar, myFun */
var myVar = 123;
function myFun() {}
```

**Comment by doberkofler** [comment]
+1

**Comment by dmose** [comment]
http://jshint.com/docs/ sez that jshint uses /* exported */, so if you choose to go with that syntax, all code that is currently using jshint gets a cheap migration path to eslint.

**Comment by rishifactual** [comment]
+1. I'd really love to have this feature in ESLint.

**Comment by nzakas** [comment]
The issue is open, so it's waiting on implementation. +1s don't matter, finding someone with the time to implement does. If you're passionate about it, please consider filing a pull request.

**Comment by xjamundx** [comment]
I'll work on this

**Comment by xjamundx** [comment]
okay starting on this today

**Comment by pixelmike** [comment]
Hi, how do I enable this?

**Comment by IanVS** [comment]
@pixelmike You'll need to use `1.0.0-rc-3`, and the docs are at http://eslint.org/docs/1.0.0/rules/no-unused-vars#exporting-variables.

**Comment by msmicker** [comment]
I can't seem to get this working - for me these two lines still give a warning:
/* exported bootstrap */
var bootstrap = require('bootstrap');

**Comment by IanVS** [comment]
What version of ESLint are you using?  Please give the output of ESLint and your config file.

**Comment by ilyavolodin** [comment]
And what environment do you have on?

**Comment by xjamundx** [comment]
I see require so I assume you're in a node environment, where exported does
not work (because your have module scope.)

If you want bootstrap globally accessible you'll need to assign it to
window.bootstrap.

Or if you just need to require() it and not assign it to any variable you
can do that as well!

**Comment by xjamundx** [comment]
@msmicker can you join the chat at https://gitter.im/eslint/eslint to discuss this more or look at https://github.com/eslint/eslint/issues/3833 for more details?

**Comment by msmicker** [comment]
Sure, thanks. I am using node/webpack. Removing the var for bootstrap worked. I recently split my project from one large one into several files. The entry file no longer references React in my own code but is 'required' for react-router. I cannot remove the var but setting to global.React worked. I am new to js and don't completely follow but that's ok :)

```
/* global require, global */

//required for boostrap
global.jQuery = require('jquery');

//required for react-router though I do not reference React in this file
global.React = require('react');

//did not need global
require('bootstrap');
```

**Comment by xjamundx** [comment]
This is getting into pure webpack advice, but you probably actually just want to do this in your webpack config, which will mean you don't even need this file.

```
	plugins: [
		new webpack.ProvidePlugin({
			$: 'jquery',
			jQuery: 'jquery',
			'window.jQuery': 'jquery'
		})
	],
```

Also react-router *does not* require a global reference to `React`...

ping me the chatroom mentioned above and i can help with this stuff....
